charts/corda-dev/values.yaml

```diff
@@ -2,55 +2,58 @@
 kafka:
   # -- enable/disable kafka.
   enabled: true
-  # Configuring Kafka image
+  # Configuring Kafka image.
   image:
-    tag: 3.2.0-debian-11-r20
-  # -- enable plaintext as a listener protocol.
-  allowPlaintextListener: true
-  # -- set a static replica count of kafka brokers.
-  replicaCount: 1
-  # -- enable/disable auto-creation of topics.
-  autoCreateTopicsEnable: false
-  # authentication configuration.
-  auth:
-    # -- protocol of the client listener.
-    clientProtocol: sasl_tls
-    # SASL configuration.
-    sasl:
-      # JAAS configuration for SASL authentication.
-      jaas:
-        # -- client users to authenticate with Kafka.
-        clientUsers:
-          - "user"
-        # -- client passwords to authenticate with Kafka; generated if not specified.
-        clientPasswords:
-    # -- protocol of the inter broker listener used for replication.
-    interBrokerProtocol: tls
-    # tls configuration.
-    tls:
-      # -- type of the tls certificates.
-      type: pem
-      # -- autogenerate a CA, and signed certificates for each broker.
-      autoGenerated: true
-  # -- the authorizer to use for access control
-  authorizerClassName: "kafka.security.authorizer.AclAuthorizer"
-  # values configuring bitnami/zookeeper dependency of bitnami/kafka.
-  zookeeper:
-    # -- set a static replica count of zookeeper nodes.
-    replicaCount: 1
+    tag: 3.6.1-debian-11-r0
+  extraConfig: "authorizer.class.name=org.apache.kafka.metadata.authorizer.StandardAuthorizer\nauto.create.topics.enable=false\nallow.everyone.if.no.acl.found=true\nsuper.users=User:controller_user;User:bootstrap"
+  # listeners configuration.
+  listeners:
+    # client configuration.
+    client:
+      # -- client protocol.
+      protocol: SASL_SSL
+      # -- client ssl authentication e.g. required.
+      sslClientAuth: none
+  # sasl configuration.
+  sasl:
+    # client configuration.
+    client:
+      # -- client users to authenticate with Kafka.
+      users:
+        - "bootstrap"
+        - "crypto"
+        - "db"
+        - "flow"
+        - "flowMapper"
+        - "verification"
+        - "membership"
+        - "p2pGateway"
+        - "p2pLinkManager"
+        - "persistence"
+        - "tokenSelection"
+        - "rest"
+        - "uniqueness" 
```

Comment:
As with PG, does changing the default set of users here cause problems? Is there another user that is always being created as I see that `values-prereqs.yaml` is using `admin` and a field from the key called `admin-password`. If that is still still being created, maybe we don't need to make the `bootstrap` user a super user after all.

Reply:
The admin-password secret isn't, but inside the kafka-prereqs-user-passwords secret, there is a key called system-user-password. This password is the same password given to the client user first in the list. Removing the `User:bootstrap` from the config doesn't affect the deployment, so it's just a case of whether we want a) want to keep creating the worker users, b) only create an admin user, or c) create an admin user and all the worker users (in this case, the bootstrap user won't have admin permissions. Incidentally this doesn't seem to impact the deployment).

Reply:
In which case, it sounds like what you have here is fine. The secret name has been updated so the likes of `values-prereqs.yaml` will have to change anyway so it doesn't matter that the user name has changed. And it also doesn't matter that you have multiple users defined if it can still just pick out the password which goes with the first one without having to parse the comma-separated list of password.